File: src/GameException.cs
using System.Runtime.Serialization;

namespace Statesman
{
    [Serializable]
    public class GameException : Exception
    {
        public GameException() { }
        public GameException(string message)
            : base(message) { }
        public GameException(string message, Exception inner)
            : base(message, inner) { }
        protected GameException(SerializationInfo info, StreamingContext context)
            : base(info, context) { }
    }
}

File: src/statesman/GameException.java
package statesman;

public class GameException extends Exception {

    private static final long serialVersionUID = -8669659145442902322L;

    public GameException() {
        super();
    }

    public GameException(String message) {
        super(message);
    }

    public GameException(Throwable cause) {
        super(cause);
    }

    public GameException(String message, Throwable cause) {
        super(message, cause);
    }

    public GameException(String message, Throwable cause,
            boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }

}
